File: Backend_IPCA_Gym/LayerBOL/Models/Ginasio.cs
namespace LayerBOL.Models
{
    public class Ginasio
    {
        /// <summary>
        /// Id do ginásio
        /// </summary>
        /// <example>1</example>
        public int id_ginasio { get; set; }
        /// <summary>
        /// Nome da instituição/estabelecimento
        /// </summary>
        /// <example>UMinho</example>
        public string instituicao { get; set; }
        /// <summary>
        /// Estado do ginásio (Ativo ou Inativo)
        /// </summary>
        /// <example>Ativo</example>
        public string estado { get; set; }
        /// <summary>
        /// Foto do ginásio que poderá ser nula
        /// </summary>
        /// <example>C:\OneDrive\ginasio.png</example>
        public string? foto_ginasio { get; set; }
        /// <summary>
        /// Contacto do ginásio
        /// </summary>
        /// <example>911922933</example>
        public int contacto { get; set; }

        /// <summary>
        /// Numero de pessoas a utilizar o ginasio
        /// </summary>
        /// <example>12</example>
        public int lotacao { get; set; }
        /// <summary>
        /// Numero de pessoas maximo que pode utilizar o ginasio
        /// </summary>
        /// <example>50</example>
        public int lotacaoMax { get; set; }
    }
}

File: IpcaGym2/app/src/main/java/vough/example/ipcagym/data_classes/Ginasio.kt
package vough.example.ipcagym.data_classes

import org.json.JSONObject

class Ginasio {
    var id_ginasio: Int? = null
    var instituicao: String? = null
    var estado : String? = null
    var foto_ginasio : String? = null
    var contacto  : Int? = null
    var lotacao  : Int? = null
    var lotacaoMax  : Int? = null

    constructor(
        id_ginasio: Int?,
        instituicao: String?,
        estado : String?,
        foto_ginasio : String?,
        contacto : Int?,
        lotacao : Int?,
        lotacaoMax : Int?

    ) {
        this.id_ginasio = id_ginasio
        this.instituicao = instituicao
        this.estado  = estado
        this.foto_ginasio  = foto_ginasio
        this.contacto  = contacto
        this.lotacao  = lotacao
        this.lotacaoMax  = lotacaoMax
    }

    fun toJson() : JSONObject {
        val jsonObj = JSONObject()

        jsonObj.put("id_ginasio", id_ginasio)
        jsonObj.put("instituicao", instituicao)
        jsonObj.put("estado", estado)
        jsonObj.put("foto_ginasio", foto_ginasio)
        jsonObj.put("contacto", contacto)
        jsonObj.put("lotacao", lotacao)
        jsonObj.put("lotacaoMax", lotacaoMax)

        return jsonObj
    }

    companion object{
        fun fromJson(jsonObject: JSONObject) : Ginasio {
            return Ginasio(
                jsonObject.getInt("id_ginasio"),
                jsonObject.getString("instituicao"),
                jsonObject.getString("estado"),
                jsonObject.getString("foto_ginasio"),
                jsonObject.getInt("contacto"),
                jsonObject.getInt("lotacao"),
                jsonObject.getInt("lotacaoMax")
            )
        }
    }
}